ethereum/contracts/Bridge.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "./Decoder.sol";
+import "./Application.sol";
+
+contract Bridge {
+    using Decoder for bytes;
+
+    uint64 public mostRecentBlock;
+    mapping(uint64 => uint64) public mostRecentBlockEvent;
+    mapping(address => bool) public applications;
+
+
+    constructor(address[] memory apps) public {
+        mostRecentBlock = 0;
+        for(uint256 i = 0; i < apps.length; i++) {
+            if(verifyApp(apps[i])) {
+                applications[apps[i]] = true;
+            }
+        }
+    }
+
+    /**
+     * @dev routes the message to the specified application ID after verifying the operator's signature
+     * @param _message address _message expected type: Message { AppID [32]byte, Payload []byte }
+     * @param _verification bytes _verification expected type: Verification { BlockNumber int64, EventID int64 }
+     */
+    function handle(bytes memory _message, bytes memory _verification)
```

Comment:
The substrate side doesn't embed the app_id within the message.

To better mirror the substrate side, the method signature would look like this:
```
submit(address _app_id, bytes memory _message, bytes memory _verification)
```

On the substrate side the signature looks like the following:
```
submit(app_id: AppID, message: Message)

struct Message {
	payload: Vec<u8>,
	verification: VerificationInput,
}
```

I don't think we need to worry about making the signatures exactly the same. So passing message and verification as separate parameters like you're doing is totally fine for this milestone.